app/form_models/offender_sar_case_form.rb

```diff
@@ -150,22 +67,14 @@ def set_empty_value_if_unset(params, field)
     params.merge!(field => "") unless params[field].present?
   end
 
+  def set_empty_value_if_unset_for_date(params, field)
+    params.merge!(field => "") unless params["#{field}_yyyy"].present? && 
+                                params["#{field}_mm"].present? && 
+                                params["#{field}_dd"].present?
+  end
```

Comment:
how did you know that? yes -_-!,  I was trying to get the date_of_birth to work, somehow in the session data I can only get the values of 'virtual fields'. not the real field, then after params_for_step() call, it will insert a empty date_of_birth which cause the validation failure :(.